### PR TITLE
[google_sign_in] Adopt code excerpts in README

### PR DESCRIPTION
**Pull request by mike-v2**
Improves README example and updates it to use code excerpts.

Part of [flutter/flutter#102679](https://github.com/flutter/flutter/issues/102679)

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

**Comment by mike-v2** [comment]
I was able to get the code excerpts from the `example/lib/main.dart` file, but they didn't always appear verbatim. Let me know if the differences outweigh the benefit of not creating an extra file, or if I should tweak the example file to better match the original README snippets.
If you do want me to create a separate file, please advise on how to structure the tests, since it seems the README code is mainly making API calls.

File: packages/google_sign_in/google_sign_in/README.md [hunk 1]
```diff
@@ -66,25 +66,32 @@ To use this plugin, follow the
 
 Add the following import to your Dart code:
 
+<?code-excerpt "example/lib/main.dart (Import)"?>
 ```dart
 import 'package:google_sign_in/google_sign_in.dart';
 ```
```
Comment (stuartmorgan) [inline packages/google_sign_in/google_sign_in/README.md:72]: We can just remove from 67 to here; this information is in the installation page instructions.

File: packages/google_sign_in/google_sign_in/example/lib/main.dart [hunk 2]
```diff
@@ -9,12 +9,15 @@ import 'dart:convert' show json;
 
 import 'package:flutter/foundation.dart';
 import 'package:flutter/material.dart';
+// #docregion Import
```
Comment (stuartmorgan) [inline packages/google_sign_in/google_sign_in/example/lib/main.dart:12]: (And remove this.)

File: packages/google_sign_in/google_sign_in/example/lib/main.dart [hunk 3]
```diff
@@ -59,7 +63,9 @@ class _SignInDemoState extends State<SignInDemo> {
       bool isAuthorized = account != null;
       // However, in the web...
       if (kIsWeb && account != null) {
+// #docregion CanAccessScopes
```
Comment (stuartmorgan) [inline packages/google_sign_in/google_sign_in/example/lib/main.dart:66]: Let's put this just above line 62 instead; it's all relevant to what the example is talking about.

File: packages/google_sign_in/google_sign_in/example/lib/main.dart [hunk 4]
```diff
@@ -59,7 +63,9 @@ class _SignInDemoState extends State<SignInDemo> {
...
         isAuthorized = await _googleSignIn.canAccessScopes(scopes);
+// #enddocregion CanAccessScopes
```
Comment (stuartmorgan) [inline packages/google_sign_in/google_sign_in/example/lib/main.dart:68]: And then this will move down one line.

**Review by stuartmorgan — APPROVED** [review]
LGTM with some minor changes, thanks!

(Several of these are actually not very good code, like the global sign-in object, but not-great code that we know compiles is still an improvement!)

**Comment by stuartmorgan** [comment]
@tarrinneal for second review once the changes are made.

**Review by tarrinneal — APPROVED** [review]
one nit

**Pull request merged**